src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -976,28 +978,16 @@ private void ExecuteSynchronized(Action ioOperation)
         {
             var configFilePath = ConfigFilePath;
 
-            // Global: ensure mutex is honored across TS sessions
-            using (var mutex = new Mutex(false, "Global\\" + EncryptionUtility.GenerateUniqueToken(configFilePath)))
+            var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
+
+            var task = ConcurrencyUtilities.ExecuteWithFileLocked(configFilePath, token =>
             {
-                var owner = false;
-                try
-                {
-                    // operations on NuGet.config should be very short lived
-                    owner = mutex.WaitOne(TimeSpan.FromMinutes(1));
-                    // decision here is to proceed even if we were not able to get mutex ownership
-                    // and let the potential IO errors bubble up. Reasoning is that failure to get
-                    // ownership probably means faulty hardware and in this case it's better to report
-                    // back than stop responding
-                    ioOperation();
-                }
-                finally
-                {
-                    if (owner)
-                    {
-                        mutex.ReleaseMutex();
-                    }
-                }
-            }
+                ioOperation();
+                return Task.FromResult(0);
+            },
+            cts.Token).ConfigureAwait(false);
+
+            task.GetAwaiter().GetResult();
```

Comment:
we can also make a sync version of the concurrency utilities, I'd hate for this to lock us up unexpectadly later.

Question: Why `Task.GetAwaiter().GetResult()` instead of `task.Result` ?